Add latch argument to advetiseImage and advertiseCamera

garaemon opened the pull request:
* Add latch argument to advetiseImage and advertiseCamera instead of
  reading latch parameter from ros server in order to set different latch
  parameter for each publisher in one nodelet.

k-okada commented:
@garaemon thank you, may be you need to change variabel name.

```
Errors << jsk_topic_tools:make /home/travis/ros/ws_jsk_common/logs/jsk_topic_tools/build.make.000.log
In file included from /home/travis/ros/ws_jsk_common/src/jsk_common/jsk_topic_tools/src/connection_based_nodelet.cpp:37:0:
/home/travis/ros/ws_jsk_common/src/jsk_common/jsk_topic_tools/include/jsk_topic_tools/connection_based_nodelet.h: In member function 'image_transport::CameraPublisher jsk_topic_tools::ConnectionBasedNodelet::advertiseCamera(ros::NodeHandle&, const string&, int, bool)':
/home/travis/ros/ws_jsk_common/src/jsk_common/jsk_topic_tools/include/jsk_topic_tools/connection_based_nodelet.h:317:12: error: declaration of 'bool latch' shadows a parameter
       bool latch;
            ^
make[2]: *** [CMakeFiles/jsk_topic_tools.dir/src/connection_based_nodelet.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs...
```

garaemon commented:
@k-okada Thanks, I fixed it.

And I added the new `advertise` method which exposes latch boolean as function argument rather than reading it from ros parameter server just like `advertiseImage` and `advertiseCamera`.
